### ISSUE TITLE
Sharing via public link broken

### ISSUE THREAD

**Issue by T0mWz** [SURFnet/rd-sram-integration#198]
When the OpenCloudMesh and FederatedGroup app are enabled and the required config parameters are set, sharing via public link is broken;
```
{"reqId":"ZIg9yIqmcMbyWLacjZtdpQAAAAY","level":3,"time":"13\/Jun\/2023:11:58:33","remoteAddr":"10.234.0.3","user":"surfsara","app":"PHP","method":"POST","url":"\/ocs\/v2.php\/apps\/files_sharing\/api\/v1\/shares?format=json","message":"Error: Call to a member function getNode() on null at \/var\/www\/owncloud\/lib\/private\/Share20\/Manager.php#842"}
{"reqId":"ZIg9yIqmcMbyWLacjZtdpQAAAAY","level":1,"time":"13\/Jun\/2023:11:58:33","remoteAddr":"10.234.0.3","user":"surfsara","app":"core","method":"POST","url":"\/ocs\/v2.php\/apps\/files_sharing\/api\/v1\/shares?format=json","message":"cleaning 1 stray shared locks for files\/42e3b9f8db08958d0d04e3f8e2cb9374"}
{"reqId":"ZIg9yIqmcMbyWLacjZtdpQAAAAY","level":1,"time":"13\/Jun\/2023:11:58:33","remoteAddr":"10.234.0.3","user":"surfsara","app":"core","method":"POST","url":"\/ocs\/v2.php\/apps\/files_sharing\/api\/v1\/shares?format=json","message":"cleaning 1 stray shared locks for files\/df7874091b33ff621e907e1ec3970e79"}
{"reqId":"ZIg9yIqmcMbyWLacjZtdpQAAAAY","level":1,"time":"13\/Jun\/2023:11:58:33","remoteAddr":"10.234.0.3","user":"surfsara","app":"core","method":"POST","url":"\/ocs\/v2.php\/apps\/files_sharing\/api\/v1\/shares?format=json","message":"cleaning 1 stray shared locks for files\/5886cc0eef51da3c2b057c42de50bb0b"}
```

**Comment by yasharpm** [comment]
I just confirmed. This issue does not exist if only the OCM app is installed. It must be something related to the federated groups app.

**Comment by T0mWz** [comment]
`An error occurred while updating share state: wrong share ID, share doesn't exist.`

**Comment by navid-shokri** [comment]
> When a federated group share is created. On remote site, the share can't be accepted. The error is shown; `An error occurred while updating share state: wrong share ID, share doesn't exist.`

It is not related to this Issue. 
this issue is resolved on this commit 38e60609720a19851c676e9caf6fd66f08921d54

the test is created and changed to green :
https://github.com/SURFnet/rd-sram-integration/blob/38e60609720a19851c676e9caf6fd66f08921d54/federatedgroups/tests/MixedGroupShareProviderTest.php#L279
